src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -2493,13 +2494,19 @@ bool LibraryCallKit::inline_vector_convert() {
     if (is_mask && is_floating_point_type(elem_bt_from)) {
       new_elem_bt_from = elem_bt_from == T_FLOAT ? T_INT : T_LONG;
     }
+    if (is_mask && is_floating_point_type(elem_bt_to)) {
+      new_elem_bt_to = elem_bt_to == T_FLOAT ? T_INT : T_LONG;
+    }
     int cast_vopc = VectorCastNode::opcode(new_elem_bt_from, !is_ucast);
-    // Make sure that cast is implemented to particular type/size combination.
-    if (!arch_supports_vector(cast_vopc, num_elem_to, elem_bt_to, VecMaskNotUsed)) {
+
+    // Make sure that vector cast is implemented to particular type/size combination.
+    bool no_vec_cast_check = is_mask &&
+                             ((src_type->isa_vectmask() && dst_type->isa_vectmask()) ||
+                              type2aelembytes(elem_bt_from) == type2aelembytes(elem_bt_to));
```

Comment:
Same check is existing at https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/vectorIntrinsics.cpp#L2551, we can do some factorization.

Reply:
Yes, we can. But consider these codes will be optimized out in the follow-up unify patch (see https://github.com/openjdk/jdk/pull/10192), do you think it's necessary to do some simple factorization here? If it's necessary, I can change the codes. Thanks!